src/HotChocolate/Utilities/src/Utilities/Cache.cs

```diff
@@ -26,18 +28,25 @@ public sealed class Cache<TValue>
     /// <param name="capacity">
     /// The maximum number of items that can be stored in this cache.
     /// </param>
+    /// <param name="diagnostics">
+    /// The diagnostics for the cache.
+    /// </param>
     /// <exception cref="ArgumentOutOfRangeException">
     /// Thrown when the capacity is less than 10.
     /// </exception>
-    public Cache(int capacity = 256)
+    public Cache(int capacity = 256, CacheDiagnostics? diagnostics = null)
     {
         ArgumentOutOfRangeException.ThrowIfLessThan(capacity, 10);
+
         _capacity = capacity;
         _ring = new CacheEntry[capacity];
         _map = new ConcurrentDictionary<string, CacheEntry>(
             concurrencyLevel: Environment.ProcessorCount,
             capacity: _capacity,
             comparer: StringComparer.Ordinal);
+        _diagnostics = diagnostics ?? NoOpCacheDiagnostics.Instance;
+        _diagnostics.RegisterCapacityGauge(() => _capacity);
```

Comment:
I am not sure if this should be a gauge - the capacity is static isnt it?

Reply:
there isnt a change over time

Reply:
yes ... but If your monitoring stack can easily compute expressions across streams like Grafana or Prometheus the two‑gauge approach is a bit cleaner because capacity stays numeric.